app/screens/home/bottomTabs/Activities.tsx

```diff
@@ -384,103 +456,331 @@ const Activities = ({ route, navigation }: any): any => {
       </ArticleHeading>
     )
   });
+  const [searchIndex, setSearchIndex] = useState<any>(null);
+  const suffixes = (term: any, minLength: any): any => {
+    if (term == null) { return []; }
+    const tokens = [];
+    for (let i = 0; i <= term.length - minLength; i++) {
+      tokens.push(term.slice(i));
+    }
+    return tokens;
+  }
+  const preprocessActivities = (activities: any): any => {
+    return activities.map((activity: any) => ({
+      ...activity,
+      normalizedTitle: activity.title,
+      normalizedSummary: activity.summary,
+      normalizedBody: activity.body
+    }));
+  };
+  // add minisearch on active child article data 
+  useEffect(() => {
+    async function initializeSearchIndex() {
+      await new Promise(resolve => setTimeout(resolve, 0));
+      const processedActivities = preprocessActivities(ActivitiesData);
+      const searchIndex = new MiniSearch({

```

Comment:
Rename searchIndex to something else to avoid conflict between two variables with the same name.